### ISSUE TITLE
Install/Setup Directions?

### ISSUE THREAD

**Issue by NonCreature0714** [kward/shunit2#93]
This is related to documentation. I'm guessing this is obvious to everyone who uses shunit2, but how do I setup shunit2 in a project for use? Do I clone it? Curl it?

**Comment by github-pawo** [comment]
Copy shunit2 script in your test folder and try the example in the quickstart section -> https://github.com/kward/shunit2#-quickstart

**Comment by EvgeniGordeev** [comment]
The easiest solution for this issue is to place `shunit2` to `Release assets` and provide the install command like (assuming the next version is 2.1.8):

```
curl -L https://github.com/kward/shunit2/releases/download/2.1.8/shunit2 -o /usr/local/bin/shunit2
chmod +x /usr/local/bin/shunit2
```

Then the quickstart example  can be transformed into:

```
#! /bin/sh
# file: examples/equality_test.sh

testEquality() {
  assertEquals 1 1
}

# Load shUnit2.
source shunit2
```

**Comment by sirosen** [comment]
I found this difficult/confusing as well, and opened a duplicate (sorry! closed now: #128 ). But there are some things in there I don't want lost:

1. Install instructions should cover including shunit2 in-tree

I want to write a testsuite and then provide my coworkers with instructions to run it. I'd like to give them the testsuite _with_ the version of shunit2 which it uses.
I think this is a reasonable/common usage -- in modern languages, I'd provide a package spec of some kind which specifies the version of a testing package to use. The only way to do that here is to include it in-tree.

2. Example scripts only source `shunit2`, but you can also invoke `shunit2` on a test script

I'm not clear on what the difference in behavior is. Any guidance? Is one preferred?

3. Test file discovery and writing a custom test runner is non obvious

If you're using `shunit2`, it's probably because you want to write a large number of tests, and maintenance will be easier if you keep them organized by subject/type/etc in separate files.
But... how do you do that? As far as I can tell, you have to write a wrapper script that does whatever discovery you want for test files, and invokes them correctly. Is that correct? Are there best practices? What's a good minimal test runner?

<details><summary>a small test runner, as an example</summary>

Is this good? Bad? Do other people have ways they prefer to do it?

```shell
#!/bin/bash
cd "$(dirname "$0")"
runsuite() {
  echo -e "\n-- Running suite: $1 --"
  bash ./lib/shunit2 "$1"
}
main() {
  if [ $# -gt 0 ]; then
    bash ./lib/shunit2 "$@"
  else
    for f in ./suites/*_test.sh; do
      runsuite "$f"
    done
  fi
}
main "$@"
```
</details>

**Comment by codingjoe** [comment]
Here, this simple setup should help people to get started in a blink:

```bash
#!/usr/bin/env bash

# … your tests

command -v shunit2 || {
  curl -sLo /usr/local/bin/shunit2 https://raw.githubusercontent.com/kward/shunit2/master/shunit2
  chmod +x /usr/local/bin/shunit2
}
source shunit2
```

**Comment by davidkhala** [comment]
> ```shell
> curl -sLo /usr/local/bin/shunit2 https://raw.githubusercontent.com/kward/shunit2/master/shunit2
>   chmod +x /usr/local/bin/shunit2
> ```

For people add this snippet but seeing error like

`chmod: cannot access '/usr/local/bin/shunit2': No such file or directory`
add `sudo` before `curl`

**Comment by codingjoe** [comment]
@davidkhala good point, I tested this on docker without `sudo`. Of course, we could simply install `shunit2` is an unprivileged location or simply download it into `/tmp` and source it from there.

**Comment by Peter-Malcolm** [comment]
Still having some issues getting this to install reliably. 

* The releases here https://github.com/kward/shunit2/releases  
only contain a tar.gz or zip containing the full source. 
* apt-get only hosts version 2.1.6 as far as I can tell. 

* I can't find a way of getting the version reliably from the program itself. 
shunit2 -v and shunit2 --version don't do what I would expect. 

* Is shunit2 safe to install reliably using? 
curl -L https://raw.githubusercontent.com/kward/shunit2/master/shunit2
If there are any dependencies on any other files in the repo presumably this will break?

For stability, I'm inclined to suggest a git clone + chmod based approach:
git clone --branch v2.1.8 https://github.com/kward/shunit2
chmod +x shunit2/shunit2
ln -s "$(realpath shunit2/shunit2)" tests/shunit2

**Comment by opsb** [comment]
There's a brew package available now
```
$ brew install shunit2
```

**Comment by HaraldKi** [comment]
My main issue with "installation" was: is the script `shunit2` self contained or why do we have this `lib` folder around? And most people these days are completely brainwashed with regard to "installation". It has to be complicated and you need package managers and all that. So I suggest to add the following to the `README.md`, after experimenting for 2 minutes (so correct as needed):

# Installation
The script `shunit2` is a self contained library, not a program to call. The most basic way to use it:
- Copy it to a place you like, maybe even to your project directory.
- Write your test script using functions provided by `shunit2`, like `assertEquals` and the like.
- In the last line of your script, invoke `source path/to/shunit2`
- Run your test script.
- Be happy. 😄
